Week3/Mohammad-Darwesh-homework-JS1-week3/mohammaddarwesh.js

```diff
@@ -0,0 +1,260 @@
+        return "input error!!! try vehicle('color',1 or 2, number >= 0)";
+}
+console.log(vehicle("red", 1, 0));
+console.log(vehicle("blue", 2, 0));
+console.log(vehicle("red", 1, 3));
+console.log(vehicle("blue", 2, 5));
+console.log(vehicle("blue", 2, -5));
+console.log(vehicle("blue", 5, 7));
+
+// task 7:
+const vehicles = ["motorbike","caravan","bike"];
+console.log(vehicles);
+
+// task 8:
+console.log(vehicles[2]);
+
+// task 9:
+function vehicle2 (color, x, age){
+    if (x === 1 && age > 0){
+    return  `a ${color} used ${vehicles[0]}`;
+    }
+    else if (x === 2 && age > 0){
+        return `a ${color} used ${vehicles[1]}`;
+    }
+    else if (x === 1 && age === 0){
+        return  `a ${color} new ${vehicles[0]}`;
+    }
+    else if (x === 2 && age === 0){
+        return `a ${color} new ${vehicles[1]}`;
+    }
+    else if (x === 3 && age > 0){
+        return `a ${color} used ${vehicles[2]}`;
+    }
+    else if (x === 3 && age === 0){
+        return  `a ${color} new ${vehicles[2]}`;
+    }
+    else 
+        return "input error!!! try vehicle('color',1 to 3, number >= 0)";
+}
+console.log(vehicle2("red", 1, 0));
+console.log(vehicle2("blue", 2, 0));
+console.log(vehicle2("blue", 3, 0));
+console.log(vehicle2("red", 1, 3));
+console.log(vehicle2("blue", 2, 5));
+console.log(vehicle2("blue", 3, 5));
+console.log(vehicle2("blue", 2, -5));
+console.log(vehicle2("blue", 5, 7));
+
+// task 10:
+function typingAdvertisement(){
+          let s = "";
+          for (let i = 0; i < vehicles.length - 2; i++) {
+               s += `${vehicles[i]}s, `;
+               // or :
+               //  s += vehicles[i]+ "s, ";
+          }
+          const advertisement = `Amazing Joe's Garage, we service ${s}${vehicles[vehicles.length - 2]}s and ${vehicles[vehicles.length - 1]}s.`;
+          // or :
+          // const advertisement = "Amazing Joe's Garage, we service " + s + "and " + vehicles[vehicles.length - 1] + "s.";
+          console.log(advertisement); 
+}
+typingAdvertisement();
+
+// task 11:
+vehicles.push("car");
+console.log(vehicles);
+console.log(vehicles.length);
+typingAdvertisement();
+
+// task 12:
+const myInformation1 = {};
+console.log(myInformation1);
+
+// task 13 & 14:
+const myTeachers = {
+    t1: "philip",
+    t2: "rop",
+    t3: "unmeshjoshi",
+    t4: "yash"
+};
+console.log(myTeachers);
+myTeachers.languages = ["html","css","cli","JS"];
+console.log(myTeachers);
+
+// task 15:
+let x = [1, 2, 3];
+let y = [1, 2, 3];
+let z = y;
+console.log(x);
+console.log(y);
+console.log(z);
+console.log(x == y);
+console.log(x === y);
+console.log(z == y);
+console.log(z == x);
+console.log(z === x);
+console.log(z === y);
+function same(a,b,c){
+    let same = a.length === b.length && a.length === c.length && b.length === c.length,
+        result,
+        len;
+    if(!same) { result = false }
+    else {
+        result = true;
+        len = a.length;
+        for(let i=0;result !== false && i < len; i++) {
```

Comment:
You assigned `true` to result on line 211, so no need to check if its true here.